src/pages/Dashboard/Modals/SupplyWithdraw/SupplyWithdrawForm.tsx

```diff
@@ -0,0 +1,201 @@
+      })
+        .times(tokenPrice)
+        .times(collateralFactor);
+      const temp = newBalance(amountInUsd);
+      setNewBorrowLimit(BigNumber.maximum(temp, 0));
+      setNewBorrowPercent(userTotalBorrowBalance.div(temp).times(100));
+    } else if (BigNumber.isBigNumber(userTotalBorrowLimit)) {
+      setNewBorrowLimit(undefined);
+      setNewBorrowPercent(userTotalBorrowBalance.div(userTotalBorrowLimit).times(100));
+    }
+  }, [amount, asset?.id, userTotalBorrowBalance, userTotalBorrowLimit]);
+
+  // Convert wallet/ supply balance into wei
+  const maxInputWei = React.useMemo(
+    () => convertCoinsToWei({ value: maxInput, tokenSymbol: assetId }),
+    [maxInput.toString()],
+  );
+  return (
+    <>
+      <TokenTextField
+        name="amount"
+        tokenSymbol={assetId as TokenId}
+        value={amount}
+        onChange={amt => setFieldValue('amount', amt, true)}
+        maxWei={maxInputWei}
+        rightMaxButtonLabel={t('supplyWithdraw.max').toUpperCase()}
+        css={styles.input}
+      />
+      <Typography component="span" variant="small1" css={styles.greyLabel}>
+        <Trans i18nKey="supplyWithdraw.walletBalance">
+          <Typography component="span" variant="small1" css={styles.whiteLabel}>
+            {{ amount: asset.walletBalance, symbol: assetId?.toUpperCase() }}
+          </Typography>
+        </Trans>
+      </Typography>
+      <InfoRow info={tokenInfo} css={styles.infoRow} className="info-row" />
+      <Delimiter />
+      <div css={styles.progressSection}>
+        <div css={styles.totalAndLimit}>
+          <Tooltip
+            placement="bottom"
+            title={
+              <Trans i18nKey="supplyWithdraw.currentBorrowBalance">
+                {{
+                  amount: currencyFormatter(userTotalBorrowBalance),
+                  percent: newBorrowPercent.toFixed(0),
+                }}
+              </Trans>
+            }
+          >
+            <Typography component="span" variant="body1" css={styles.greyLabel}>
+              <Trans i18nKey="supplyWithdraw.currentAmount">
+                <Typography component="span" variant="body1" css={styles.whiteLabel}>
+                  {{ amount: currencyFormatter(userTotalBorrowBalance) }}
+                </Typography>
+              </Trans>
+            </Typography>
+          </Tooltip>
+          <Tooltip
+            placement="bottom"
+            title={
+              <Trans i18nKey="supplyWithdraw.maxBorrowTooltip">
+                {{ amount: currencyFormatter(userTotalBorrowLimit) }}
+              </Trans>
+            }
+          >
+            <Typography component="span" variant="body1" css={styles.greyLabel}>
+              <Trans i18nKey="supplyWithdraw.maxAmount">
+                <Typography component="span" variant="body1" css={styles.whiteLabel}>
+                  {{ amount: currencyFormatter(userTotalBorrowLimit) }}
+                </Typography>
+              </Trans>
+            </Typography>
```

Comment:
I am nesting the typography tag to add different styles. I chose to use typography so that we can leverage the theme as opposed to using dom tags directly. 

Per a previous discussion I am using Typography exclusively for rendering text.

Reply:
Got you.

Same thing here though: the whole sentence needs to be rendered (not just an object).